javascript_nodejs/14.nlp-with-dispatch/dialogs/homeAutomation/home-automation-state.js

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const HOME_ATUOMATION_STATE_PROPERTY = 'homeAutomation.state';
+
+class DeviceState {
+    /**
+     * 
+     * @param {String} device 
+     * @param {String} room 
+     * @param {String} deviceProperty 
+     * @param {String} deviceState 
+     */
+    constructor(device, room, deviceProperty, deviceState) { 
+        this.deviceName = device?device:'';
+        this.room = room?room:'';
+        this.deviceProperty = deviceProperty?deviceProperty:'';
+        this.deviceState = deviceState?deviceState:'';
+    }
+};
+
+class HomeAutomationState {
+    /**
+     * 
+     * @param {object} convoState Conversation state
+     * @param {object} userState User state 
+     */
+    constructor(convoState, userState) {
+        if(!convoState || !convoState.createProperty) throw('Invalid conversation state provided.');
+        if(!userState || !userState.createProperty) throw('Invalid user state provided.');
+
+        // device property accessor for home automation scenario.
```

Comment:
"Device". If the comments ends in a ".", it's a sentence.

Reply:
Fixed.